components/playground/instance/process.go

```diff
@@ -55,27 +68,34 @@ func (p *Process) setOutputFile(fname string) error {
 	return nil
 }
 
-func (p *Process) setOutput(w io.Writer) {
+func (p *process) setOutput(w io.Writer) {
 	p.cmd.Stdout = w
 	p.cmd.Stderr = w
 }
 
-// NewProcess create a Process instance.
-func NewProcess(ctx context.Context, dir string, name string, arg ...string) *Process {
+func (p *process) Cmd() *exec.Cmd {
+	return p.cmd
+}
+
+// NewComponentProcess create a Process instance.
+func NewComponentProcess(ctx context.Context, dir, binPath, component string, version v0manifest.Version, arg ...string) (Process, error) {
 	if dir == "" {
 		panic("dir must be set")
 	}
 
-	cmd := exec.CommandContext(ctx, name, arg...)
-	cmd.Env = append(
-		os.Environ(),
-		fmt.Sprintf("%s=%s", localdata.EnvNameInstanceDataDir, dir),
-	)
+	env, err := environment.InitEnv(repository.Options{})
+	if err != nil {
+		return nil, err
+	}
 
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
+	oldDir := os.Getenv(localdata.EnvNameInstanceDataDir)
+	defer os.Setenv(localdata.EnvNameInstanceDataDir, oldDir)
```

Comment:
I think the better way to pass the instance data directory is to use a string slice instead of set the os environment. It's a potential bug if we call `NewComponentProcess` parallelly in the future.